### PR TITLE
Cropping feature implemented

### PR DESCRIPTION
**Pull request by JancoDVTDev**
### Free Crop
Allows user to freely select a range to crop the image

### Reset
Load original image

### Presets (To be Finilised in next feature)
- 3 Presets helps the user get a perfect square, rectangle and portrait image

### CODE REVIEW

**Comment by JancoDVTDev** [comment]
### Example

IMG_3317

**Comment by XanderSchoeman** [comment]
Just fix your Codacy status check.

**Review by XanderSchoeman — APPROVED** [review]
Pretty neat feature you got working. If the others have suggestions then yeah, looks good from my side.

File: camPod/Classes/EditPhoto/VIewModel/PhotoEditorViewModel.swift [hunk 1]
```diff
@@ -146,4 +146,104 @@ public class PhotoEditorViewModel {
...
+        let scaleFactorX = CGFloat(cgCropImage!.width)/scaledWidth
+        let scaleFactorY = CGFloat(cgCropImage!.height)/scaledHeight
+
+        if originalCGCropImageWidth > originalCGCropImageHeight {
+            scaledOriginY = (cropView.frame.origin.y - (imageViewHeight - scaledHeight)) * scaleFactorY
+            scaledOriginX = cropView.frame.origin.x * scaleFactorX
+        } else {
+            scaledOriginX = (cropView.frame.origin.x - (imageViewWidth/2 - scaledWidth!/2)) * scaleFactorX
+            scaledOriginY = (cropView.frame.origin.y - 152) * scaleFactorY
+            //VERY IMPORTANT NOTE 55 is the value from screen top to imageview also 152
```
Comment (bgouws) [inline camPod/Classes/EditPhoto/VIewModel/PhotoEditorViewModel.swift:184]: Just remember to remove comments.

File: camPod/Classes/EditPhoto/View/PhotoEditingViewController.swift [hunk 2]
```diff
@@ -58,7 +67,10 @@ import CoreImage.CIFilterBuiltins
             inEditPhotoImageView.image = fifthFilterButton.backgroundImage(for: .normal)
             //viewModel.applyPredefinedFilter(image: inEditPhoto, selectedFilter: 0)
```
Comment (bgouws) [inline camPod/Classes/EditPhoto/View/PhotoEditingViewController.swift:68]: Also here.

File: camPod/Classes/EditPhoto/View/PhotoEditingViewController.swift [hunk 3]
```diff
@@ -75,7 +87,8 @@ import CoreImage.CIFilterBuiltins
             inEditPhotoImageView.image = secondFilterButton.backgroundImage(for: .normal)
             //viewModel.applyPredefinedFilter(image: inEditPhoto, selectedFilter: 1)
```
Comment (bgouws) [inline camPod/Classes/EditPhoto/View/PhotoEditingViewController.swift:88]: Lots of comments here.

**Review by bgouws — APPROVED** [review]
Just make sure Codacy is happy

**Pull request merged**